core/json-utils/src/main/java/software/amazon/awssdk/protocols/jsoncore/internal/EmbeddedObjectJsonNode.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.protocols.jsoncore.internal;
+
+import java.util.List;
+import java.util.Map;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.protocols.jsoncore.JsonNode;
+import software.amazon.awssdk.utils.Validate;
+
+/**
+ * An embedded object {@link JsonNode}.
+ */
+@SdkInternalApi
+public final class EmbeddedObjectJsonNode implements JsonNode {
+    private final Object embeddedObject;
+
+    public EmbeddedObjectJsonNode(Object embeddedObject) {
+        Validate.paramNotNull(embeddedObject, "embeddedObject");
```

Comment:
Because these constructors are only called internal to the module, I was fine trusting that Jackson would never return invalid types. I agree that this is inconsistent, though. I'd be fine going either way, because this check is pretty much free from a performance perspective. What do you think would be better?

Reply:
Then I think it's fine if we skip the null check since it's internal and Jackson doesn't return invalid types